crates/uv-installer/src/plan.rs

```diff
@@ -264,7 +264,11 @@ impl<'a> Planner<'a> {
                     // Find the most-compatible wheel from the cache, since we don't know
                     // the filename in advance.
                     if let Some(wheel) = built_index.url(sdist)? {
-                        if wheel.filename.name == sdist.name {
+                        if wheel.filename.name == sdist.name
```

Comment:
These will be false if we don't know the sdist version, right? Should we instead maintain the old behavior in that case?

Something like:
```
if wheel.filename.name == sdist.name
  && dist.version().map_or(true, |version| version == &wheel.filename.version)
```